xls/modules/zstd/data_generator.cc

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "xls/modules/zstd/data_generator.h"
+
+#include <algorithm>
+#include <array>
+#include <cstdint>
+#include <cstdio>
+#include <filesystem>
```

Comment:
can you add a `  // NO LINT` comment here and in other place where `<filesystem>` is included?

Reply:
Is there any warning generated by `clang-tidy` for those? I don't see any after running [dev_utils/run-clang-tidy-cached.sh](https://github.com/google/xls/blob/main/dev_utils/run-clang-tidy-cached.sh)

Reply:
not yet, since it's an internal check /cc @hzeller

Reply:
Oh, ok. I added the comment to all `filesystem` include directives

Reply:
Sorry for the extra work. Yes, this is an artifact of us usually preferring a different way of dealing with file operations, so there is tooling that warns about it. But this _"usually"_ does not apply to this project of course.